data/python.gram

```diff
@@ -317,27 +362,39 @@ class Parser(Parser):
                 self._tokenizer.get_lines(list(range(start[0], end[0] + 1)))
             )
 
-        return  SyntaxError(
-            message,
-            (self.filename, start[0], start[1], line)
-        )
+        args = (self.filename, start[0], start[1] + 1, line)
+        if sys.version_info >= (3, 10):
+            args += (end[0], end[1] + 1)
+
+        return SyntaxError(message, args)
 
     def raise_raw_syntax_error(
         self,
         message: str,
         start: Optional[Tuple[int, int]] = None,
         end: Optional[Tuple[int, int]] = None
-    ) -> None:
+    ) -> NoReturn:
         raise self._build_syntax_error(message, start, end)
 
     def make_syntax_error(self, message: str) -> None:
         return self._build_syntax_error(message)
 
-    def raise_syntax_error(self, message: str) -> None:
+    def expect_forced(self, res: Any, expectation: str) -> Optional[tokenize.TokenInfo]:
```

Comment:
Where are we using this method? IIRC forced tokens already work (https://github.com/we-like-parsers/pegen/blob/e28fe4fb1972c55af5ddb6a7bdd9cba4ea072b81/tests/test_pegen.py#L589)

Reply:
This is needed to get the right error location. CPython report equal start and end for forced token which is not what we do in the default implementation. The default parser only has make_syntax_error which queries the last token and use start and end which is reasonable in general but not in this special case.

Reply:
Do we have tests covering this difference? Maybe we should add some to `test_pegen.py` to be explicit about it.

Reply:
We have 4 tests in test_syntax_error_handling.py failing if I comment this out.